.github/workflows/snapshot-publish.yml

```diff
@@ -0,0 +1,67 @@
+name: Publish in-development builds from main
+
+on:
+  schedule:
+    # Run daily on week days
+    - cron:  '0 11,23 * * 1-5'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}
+  cancel-in-progress: false
+
+jobs:
+  java:
+    name: Publish from main
+    runs-on: ubuntu-latest
+    env:
+      SPARK_LOCAL_IP: localhost
+    if: ${{ env.MAVEN_USERNAME }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup runner
+        uses: ./.github/actions/setup-runner
+      - name: Setup Java, Maven
+        uses: ./.github/actions/dev-tool-java
+
+      # Check that the commit that is becoming a release has passed CI.
+      # Note: intentionally hard-coded projectnessie/nessie, so this check even works when *testing*
+      # the workflow on a PR-branch.
+      - name: Check commit status
+        run: |
+          GIT_HEAD_SHA="$(git rev-parse HEAD)"
+          echo ${{ secrets.GITHUB_TOKEN }} | gh auth login --with-token
+          gh api repos/projectnessie/nessie/commits/${GIT_HEAD_SHA}/check-runs --jq 'if ([.check_runs[] | select(.name | endswith(" release") or startswith("Dependabot ") or startswith("codecov/") or startswith("Report ") | not ) | select(.conclusion != "skipped") | .conclusion // "pending" ] | unique == ["success"]) then "OK" else error("Commit checks are not OK") end'
```

Comment:
What if HEAD has CI in progress? Is this a concern? Could we use `gh api` to find the last commit that has complete CI results?

Reply:
Yea - CI in progress would let this check fail. Given that the wf is scheduled twice per day, that shouldn't be a big issue. And we can run it manually at any time.
Finding the last commit that's fine is probably over-engineering. It's doable, but it's probably not worth it.
Alternative could be to skip this check and assume that CI's fine.

Reply:
Fair enough :+1: